### PR TITLE
chore: prune node modules dir size in docker build

### PR DESCRIPTION
**Pull request by hazzadous**
This removes some unwanted node modules files and packages.

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

**Review by guidoiaquinti — APPROVED** [review]
Could be nice to make sure the image works but overall 👍

File: production.Dockerfile [hunk 1]
```diff
@@ -35,8 +35,7 @@ RUN apk --update --no-cache add \
     "make~=4.3" \
     "g++~=10.3" \
     "gcc~=10.3" \
-    "python3~=3.9" \
```
Comment (guidoiaquinti) [inline production.Dockerfile]: Not part of this PR but I thought we were using Python `3.8.12` (see tests, docs, etc..)

**Comment by hazzadous** [comment]
Not much of an improvement, 20MB. I'm closing but if someone else feels like taking please do, but there was a failing test and I don't want to get distracted atm cc @guidoiaquinti
